inference-engine/thirdparty/movidius/mvnc/src/mvnc_data.c

```diff
@@ -191,9 +191,14 @@ static ncStatus_t patchSetWdSwitchCommand(char **firmware, size_t *length, const
     size_t currLength = *length;
     size_t executeCommandIdx = 0;
     char executeCommandFound = 0;
-    size_t i = 0;
 
-    for (i = currLength - 1; i >= 0; i--) {
+    if (currLength < 0 || currLength > INT64_MAX) {
+        mvLog(MVLOG_ERROR, "Fail to find execute command");
```

Comment:
is this check necessary for clockwork?

Reply:
I guess yes, because there is a chance of loss in conversion from `size_t` to `int64_t`. Anyway, I don't think there is any drawback..

Reply:
`currLength` still is `size_t `. Why do we have to perform a comparison `< 0`?

Why not:
```
if(currLength == 0) {
        mvLog(MVLOG_ERROR, "Fail to find execute command");
        return NC_ERROR;
}
```